# Overview #

We present a fast and accurate algorithm, NetCoffee, which allows to find a global alignment of multiple PPI networks. NetCoffee searches for a global alignment by maximizing a target function using [simulated annealing](http://en.wikipedia.org/wiki/Simulated_annealing) (SA) on a set of weighted bipartite graphs that are constructed using a triplet approach similar to T-Coffee.

The main algorithm can be subsumed in the following four steps:

  1. building the PPI networks and a library of bipartite graphs;
  1. assigning an integrated weight to each edge in the bipartite graphs using a triplet extension approach similar to T-Coffee;
  1. building a search space that consists of candidate edges (protein pairs);
  1. simulated annealing (SA) with a large number of iterations of a Metropolis Scheme to maximize a scoring function for global alignments.

To assess its performance, NetCoffee was applied to four real datasets. Our results suggest that NetCoffee remedies several limitations of previous algorithms, outperforms all existing alignment tools in terms of speed, and nevertheless identifies biologically
meaningful alignments.


# Features #

  * It can align multiple PPI networks without additional information, such as a training dataset and a phylogenetic tree.
  * It can find a global alignment of six species within several minutes.
  * It scales to multiple networks with tens of thousands of proteins and interactions.
  * It is the first alignment tool that can run with multiple cores in parallel.



# Results #

Applied to five eukaryotic species (i.e. Dataset-2 in the paper), NetCoffee successfully recognizes a global alignment covering approximately 50% of the proteins; and about 45% of the match-sets are qualified.
For example, it aligns these five proteins, i.e. _P11171_, _Q12923_, _P46662_, _P46150_, and _Q24564_, into a match-set because their annotation of molecular function are significantly conserved as what the go-tree suggests below (this figure was generated by using [goTermfinder](http://search.cpan.org/dist/GO-TermFinder/lib/GO/TermFinder.pm)),

![https://netcoffee.googlecode.com/svn/trunk/images/GOview.3927.png](https://netcoffee.googlecode.com/svn/trunk/images/GOview.3927.png)

# Corrections #

See some error corrections of the paper [here](https://code.google.com/p/netcoffee/wiki/Corrections).


# FAQ #

See a list of frequently asked questions [here](https://code.google.com/p/netcoffee/wiki/FrequentlyAskedQuestions).

# Contact #

Please leave us feedback on your experience, especially if you find any bugs. Thanks for helping us improve NetCoffee!

Email: Jialu.Hu{at}fu-berlin.de

# Other existing tools #
[LocalAli](https://code.google.com/p/localali/)

[BEAMS](http://bioinformatics.oxfordjournals.org/content/30/4/531.full)

[SMETANA](http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0067995)

[IsoRank-N](http://bioinformatics.oxfordjournals.org/content/25/12/i253.full)

[GEDEVO-M](http://gedevo.mpi-inf.mpg.de/multiple-network-alignment/)

# Reference #

Jialu Hu, Birte Kehr, and Knut Reinert, [NetCoffee: A Fast and Accurate Global Alignment Approach to Identify Functionally Conserved Proteins in Multiple Networks, Bioinformatics (2014) 30 (4): 540-548, first published online December 13, 2013, doi:10.1093/bioinformatics/btt715](http://bioinformatics.oxfordjournals.org/content/30/4/540.abstract)